src/client/opamClient.ml

```diff
@@ -171,7 +171,7 @@ let upgrade_t
       OpamConsole.errmsg "%s\n%s\n"
         (OpamStd.Format.itemize (fun x -> x) cycles)
         "You may try upgrading packages individually to work around this."
-    end else begin
+    end else if reasons <> [] then begin
```

Comment:
I am not sure I understand this fix. If the code reaches this point, it means there are conflicts, but for some reason, `reasons` and `cycles` are both empty 🤔 

But if the error messages below are not shown, won't it be worse than the current situation? My concern is that opam will finish without upgrading, but the user has no indication of what has happened.

Reply:
ah you're right I did it too fast without really looking. Mmh, maybe that was fixed in master with https://github.com/ocaml/opam/pull/5263. Could you try simply master instead (whenever you have a reproducible case)?